projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts

```diff
@@ -210,30 +220,38 @@ export class PerformanceViewComponent implements OnInit, OnDestroy {
     if (this.includeThreadGroupChart) {
       this.createThreadGroupsChart(this.findRequest);
     }
-  }
+    this.initializationTasks.push(this.tableInitializedSubject); // we wait also for the table
+    this.initializationTasks.push(this.rangerLoadedSubject);
 
-  // we assume the settings input has been changed
-  reconstructAllCharts() {
-    this.findRequest = this.prepareFindRequest(this.settings); // we don't want to lose active filters
-    this.updateAllCharts();
-  }
-
-  mergeRequestWithActiveFilters() {
-    this.deleteObjectProperties(this.findRequest.params, this.filtersComponent.getAllFilterAttributes()); // we clean the request first
-    Object.assign(this.findRequest.params, this.executionContext.getActiveFilters());
+    forkJoin(this.initializationTasks.map((obs) => obs.pipe(take(1)))).subscribe((allCompleted) =>
```

Comment:
Instead of having the state arrays `initializationTasks`, `updateTasks`, I would suggest to the following approach.
Methods like `createSummaryChart`, `crateByStatusChart` should return observables. For example:
```
createSummaryChart(request: FindBucketRequest): Observable<unknown> {
  return this.timeSeriesService.fetchBuckets(request)
     .pipe(tap(response) => {
        this.createChart(TsChartType.OVERVIEW, request, response);
    });
}
```
So no additional subjects will require. And return observable will be **cold** (autocompleted), because it is a http requset. (In comparison subjects are **hot** observables - not autocompleted)

After that change `createAllCharts` will look like:
```
createAllCharts(): void {
   const charts$ = [
      this.createSummaryChart(this.findRequest),
      this.createByStatusChart(this.findRequest),
      this.createByKeywordsCharts(this.findRequest)
   ];
   if (this.includeThreadGroupChart) {
     charts$.push(this.createThreadGroupsChart(this.findRequest));
   }
   forkJoin(charts$).subsribe(() => this.onInitiailizationComplete.emit());
}
```